analytics.go

```diff
@@ -369,7 +369,9 @@ func (m *Message) setTimestamp(s string) {
 
 // Set message id.
 func (m *Message) setMessageId(s string) {
-	m.MessageId = s
+	if m.MessageId == "" {
```

Comment:
I think we should probably change this outside the `setMessageId` method eh? otherwise the name seems a bit misleading